From 8d67e92331835dcb93fab12fe8ac3e8614eeb4c9 Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <xiaowen.liu@freescale.com>
Date: Thu, 22 Oct 2015 17:10:56 +0800
Subject: [PATCH 568/602] MA-7147 Enable single CSI camera support on
 Sabresd-6qp board.

add ov5640 csi camera as back camera support for Sabresd-6qp board.
split back camera set for Sabresd platform.
setprop media.settings.xml to /etc/media_profiles_720p.xml for 6qp.
set opengles version to 3.0 for 6qp.

Signed-off-by: Liu Xiaowen <xiaowen.liu@freescale.com>
---
 imx6/imx6.mk                |  2 +-
 sabresd_6dq/init.i.MX6DL.rc |  6 ++++++
 sabresd_6dq/init.i.MX6Q.rc  |  6 ++++++
 sabresd_6dq/init.i.MX6QP.rc | 13 +++++++++++++
 sabresd_6dq/init.rc         |  4 ----
 5 files changed, 26 insertions(+), 5 deletions(-)
 create mode 100755 sabresd_6dq/init.i.MX6QP.rc

diff --git a/imx6/imx6.mk b/imx6/imx6.mk
index bd073eb..e478de1 100644
--- a/imx6/imx6.mk
+++ b/imx6/imx6.mk
@@ -352,7 +352,7 @@ PRODUCT_COPY_FILES +=	\
 	device/fsl/common/display/display_mode_fb4.conf:system/etc/display_mode_fb4.conf \
     device/fsl-proprietary/media-profile/media_codecs_google_audio.xml:system/etc/media_codecs_google_audio.xml \
     device/fsl-proprietary/media-profile/media_codecs_google_video.xml:system/etc/media_codecs_google_video.xml \
-
+    device/fsl-proprietary/media-profile/media_profiles_720p.xml:system/etc/media_profiles_720p.xml
 
 # we have enough storage space to hold precise GC data
 PRODUCT_TAGS += dalvik.gc.type-precise
diff --git a/sabresd_6dq/init.i.MX6DL.rc b/sabresd_6dq/init.i.MX6DL.rc
index 0113bfb..d0c5c53 100755
--- a/sabresd_6dq/init.i.MX6DL.rc
+++ b/sabresd_6dq/init.i.MX6DL.rc
@@ -1,4 +1,10 @@
 on boot
 
+    # Set OpenGLES version
+    setprop ro.opengles.version 131072
+
     # Set GPU 3D minimum clock to 8/64
     write /sys/bus/platform/drivers/galcore/gpu3DMinClock 8
+
+    # set back camera.
+    setprop back_camera_name ov5640_mipi
diff --git a/sabresd_6dq/init.i.MX6Q.rc b/sabresd_6dq/init.i.MX6Q.rc
index e463fe5..df897fa 100755
--- a/sabresd_6dq/init.i.MX6Q.rc
+++ b/sabresd_6dq/init.i.MX6Q.rc
@@ -1,4 +1,10 @@
 on boot
 
+    # Set OpenGLES version
+    setprop ro.opengles.version 131072
+
     # Set GPU 3D minimum clock to 3/64
     write /sys/bus/platform/drivers/galcore/gpu3DMinClock 3
+
+    # set back camera.
+    setprop back_camera_name ov5640_mipi
diff --git a/sabresd_6dq/init.i.MX6QP.rc b/sabresd_6dq/init.i.MX6QP.rc
new file mode 100755
index 0000000..66a34fb
--- /dev/null
+++ b/sabresd_6dq/init.i.MX6QP.rc
@@ -0,0 +1,13 @@
+on boot
+
+    # Set OpenGLES version
+    setprop ro.opengles.version 196608
+
+    # Set GPU 3D minimum clock to 3/64
+    write /sys/bus/platform/drivers/galcore/gpu3DMinClock 3
+
+    # set media_profile.xml
+    setprop media.settings.xml /etc/media_profiles_720p.xml
+
+    # set back camera.
+    setprop back_camera_name ov5640_mipi,ov5640_camera
diff --git a/sabresd_6dq/init.rc b/sabresd_6dq/init.rc
index 520ede6..2cad527 100755
--- a/sabresd_6dq/init.rc
+++ b/sabresd_6dq/init.rc
@@ -54,7 +54,6 @@ on boot
     #For landscape mode, orient is 0
     #For portrait mode, orient is 90
     #the android before honycomb are all in portrait mode
-    setprop back_camera_name ov5640_mipi
     setprop back_camera_orient 0
     setprop front_camera_name uvc,ov5642_camera,ov5640_camera
     setprop front_camera_orient 0
@@ -87,9 +86,6 @@ on boot
 #  Enable adb security for JB4.2.2
     setprop ro.adb.secure 1
 
-    # Set OpenGLES version
-    setprop ro.opengles.version 131072
-
     # Set rotation to 270 to cofigure as portrait mode
     setprop ro.sf.hwrotation 0
 
-- 
1.8.0

